src/service/storage_arango.py

```diff
@@ -210,4 +224,43 @@ async def get_collection_version_by_num(self, collection_id: str, ver_num: int
             raise errors.NoSuchCollectionVersionError(
                 f"No collection {collection_id} with version number {ver_num}")
         doc = await cur.next()
-        return models.SavedCollection.construct(**_remove_arango_keys(doc))
+        return _doc_to_saved_coll(doc)
+    
+    async def get_collection_versions(
+        self, collection_id: str, max_ver: int = None, limit: int = 1000
+    ) -> list[models.SavedCollection]:
+        """
+        List versions of a collection, sorted from highest to lowest.
+
+        collection_id - the ID of the collection.
+        max_ver - the maximum version to return. This can be used to page through the results.
+        limit - the maximum number of versions to return. Default and maximum value is 1000.
+        """
+        if not limit or limit < 1:
+            limit = 1000
+        if limit > 1000:
+            raise errors.IllegalParameterError("Limit must be <= 1000")
```

Comment:
I am okay with raise. But should we consider doing `limit = min(limit, 1000)`?

Reply:
Hmm, I'm ok with modifying a user's inputs when it's clear they're just being lazy or utterly wrong (like 0 or negative limits), but a user might not read the documentation correctly and think that specifying a higher limit is ok, and then think they got all the results. In that case I think it's better to expressly tell them the input is invalid

Reply:
👍

Reply:
That being said, the limit isn't exposed in the server api for now